app/models/bishop.rb

```diff
@@ -4,4 +4,22 @@ def move_legal?(to_row, to_col)
     col_diff = (col - to_col).abs
     row_diff == col_diff
   end
+
+  def path_to_king
+    king = game.pieces.where.not(is_black: is_black).where(type: 'King').first
+    return nil unless (king.row - row).abs == (king.col - col).abs
```

Comment:
Coming back to this, I think that using `valid_move?` is the right approach, but if it ends up that there's unintended side-effects of using `valid_move?` then lets leave the `path_to_king` methods alone.